_posts/2020-11-04-oneway-sslv2hello.adoc

```diff
@@ -0,0 +1,335 @@
+to accept ``TLSv1.2`` only. To enable ``SSLv2Hello``, we need to update our ``ssl-context`` configuration in the
+Elytron subsystem to modify the attribute ``protocols`` to include ``SSLv2Hello`` as follows:
+
+[source,shell]
+----
+/subsystem=elytron/server-ssl-context=ssl-context-aaad4a67-14f1-4869-a01c-9fc56c3ff884:write-attribute(name=protocols,value=[TLSv1,SSLv2Hello])
+
+reload
+----
+
+NOTE: When configuring any SSL context to enable ``SSLv2Hello`` (client or server), you always
+need to specify an additional SSL/TLS protocol. ``SSLv2Hello`` cannot be specified by itself,
+as it is a pseudo protocol which is only used to negotiate another SSL/TLS version to be used
+throughout the rest of the communication.
+
+=== Configure the Client's Trust Store
+
+Export the server's certificate to a file by navigating to ``WILDFLY_HOME/standalone/configuration`` and
+running the following command:
+
+[source,shell]
+----
+keytool -export -alias localhost -keystore server.keystore -file myCert.cer -storepass secret
+----
+
+Import the certificate into the client's trust store. Note the following command will create a trust store
+if it does not exist at the location specified.
+
+[source, shell]
+----
+keytool -import -file myCert.cer -alias localhost -keystore client.keystore -storepass secret
+----
+
+You should see the following output next:
+
+[source,shell]
+----
+Owner: CN=localhost, OU=wildfly, O=jboss, L=Raleigh, ST=North Carolina, C=US
+Issuer: CN=localhost, OU=wildfly, O=jboss, L=Raleigh, ST=North Carolina, C=US
+Serial number: 7af5ad9991816bbc
+Valid from: Wed Nov 04 16:15:09 EST 2020 until: Thu Nov 04 16:15:09 EDT 2021
+Certificate fingerprints:
+	 SHA1: 24:6E:96:9A:8B:D2:FF:2B:7B:58:87:84:03:F3:ED:C6:56:8B:96:5B
+	 SHA256: 27:51:F6:7A:F8:51:7D:6E:CB:DA:4A:9A:75:E5:9A:AD:06:88:1B:AE:40:73:7D:D8:E4:4F:22:CA:8E:03:94:37
+Signature algorithm name: SHA256withRSA
+Subject Public Key Algorithm: 2048-bit RSA key
+Version: 3
+
+Extensions:
+
+#1: ObjectId: 2.5.29.14 Criticality=false
+SubjectKeyIdentifier [
+KeyIdentifier [
+0000: FF BA 34 E8 94 94 BE EB   25 B5 C5 4E B3 B2 A4 34  ..4.....%..N...4
+0010: 97 EE A2 24                                        ...$
+]
+]
+
+Trust this certificate? [no]: y
+Certificate was added to keystore
+----
+
+You have now created the client's trustore and imported the server's certificate.
+
+=== Configure a Client that sends SSLv2Hello messages
+
+The example application presents a simple client ``Client.java`` which checks the connection
+to the server is successful. To ensure the client uses ``SSLv2Hello`` in its initial
+handshake, we have provided a common configuration framework in
+``wildfly-config.xml``.
+
+Review this file to see how ``SSLv2Hello`` is enabled in the client SSL context along with
+``TLSv1``.
+
+
+As a final step in configuring the client, update the path to the client's trust store to
+``WILDFLY_HOME/standalone/configuration/client.keystore``.
```

Comment:
Just a minor comment, but may be better to use `/PATH/TO/client.keystore` or something like that since the client's keystore likely won't be in the server config directory.

Reply:
Updated it to ``/PATH/TO/client.keystore``